### PR TITLE
Support input limits and `has_more` loop in containerized ingest

### PR DESCRIPTION
**Pull request by s373r**
Related issue: https://github.com/kamu-data/kamu-cli/issues/302

### CODE REVIEW

File: src/infra/core/tests/tests/ingest/test_fetch.rs [hunk 1]
```diff
@@ -637,22 +590,10 @@ async fn test_fetch_container_ok() {
...
     let fetch_step = FetchStep::Container(FetchStepContainer {
-        image: crate::utils::HttpServer::IMAGE.to_owned(),
-        command: Some(vec!["/bin/bash".to_owned()]),
-        args: Some(vec![
-            "-c".to_owned(),
-            "printf \"city,population\nA,1000\nB,2000\nC,3000\n\"".to_owned(),
-        ]),
+        image: BUSYBOX.to_owned(),
+        command: Some(vec!["printf".to_owned()]),
+        args: Some(vec!["city,population\nA,1000\nB,2000\nC,3000\n".to_owned()]),
```
Comment (sergiimk) [inline src/infra/core/tests/tests/ingest/test_fetch.rs:596]: Should we use `CSV_BATCH_OUTPUT` constant here too, now that you separated data from the command?

File: src/infra/core/tests/tests/ingest/test_fetch.rs [hunk 2]
```diff
@@ -686,6 +630,387 @@ async fn test_fetch_container_ok() {
...
+const HAS_MORE_TESTER_SCRIPT: &str = indoc! {r#"
+    #!/usr/bin/env sh
+
+    set -euo pipefail
+
+    if [[ -n "${DEBUG:-}" ]]; then
+      set -x
+    fi
+
+    ROWS_COUNT=100
```
Comment (sergiimk) [inline src/infra/core/tests/tests/ingest/test_fetch.rs:980]: I found it hard to understand the tests without closely reading this script, and it's mainly because it **hardcodes** the row count at 100.

I feel like it would be a bit simpler to understand if `ROW_COUNT` was passed as a parameter, so that in tests you could clearly see how `ROW_COUNT` compares to `ODF_BATCH_SIZE`.

File: src/infra/core/tests/tests/ingest/test_fetch.rs [hunk 3]
```diff
@@ -686,6 +630,387 @@ async fn test_fetch_container_ok() {
...
+            CSV_BATCH_OUTPUT
+        );
+
+        match res_2 {
+            FetchResult::Updated(x) => x.source_state,
+            _ => unreachable!(),
+        }
+    };
+    let prev_source_state = {
+        let fetch_step_3 = FetchStep::Container(FetchStepContainer {
```
Comment (sergiimk) [inline src/infra/core/tests/tests/ingest/test_fetch.rs:888]: To me it seems like two iterations (one expecting `has_more==true`  another `false`) is enough here. If 3rd and 4th iterations are somehow special please put a comment above them explaining what is being checked.

File: src/infra/core/tests/tests/ingest/test_fetch.rs (hunk 1 above)
Comment (s373r) [reply src/infra/core/tests/tests/ingest/test_fetch.rs]: A good catch! Yes, can be replaced -- Done

File: src/infra/core/tests/tests/ingest/test_fetch.rs (hunk 2 above)
Comment (s373r) [reply src/infra/core/tests/tests/ingest/test_fetch.rs]: It definitely makes sense -- Done

File: src/infra/core/tests/tests/ingest/test_fetch.rs (hunk 3 above)
Comment (s373r) [reply src/infra/core/tests/tests/ingest/test_fetch.rs:924]: Added comments to each step, I hope it looks like more clear -- **Q:** What do you think?

Comment (sergiimk) [reply src/infra/core/tests/tests/ingest/test_fetch.rs:924]: I think 2nd iteration doesn't test any new boundary conditions that are not covered elsewhere and could be removed to make the test shorter, but I'm fine with it either way.

**Review by zaychenko-sergei — APPROVED** [review]
Plz sync with the latest master
